akka-projection-core-test/src/test/scala/akka/projection/ProjectionBehaviorSpec.scala

```diff
@@ -248,6 +239,37 @@ class ProjectionBehaviorSpec extends ScalaTestWithActorTestKit with AnyWordSpecL
 
     }
 
+    "restart previously stopped projection" in {
+
+      val testProbe = testKit.createTestProbe[ProbeMessage]()
+
+      // we create a source from a publisher so to have fine grained control over the production of elements
+      val testPublisher = TestPublisher.probe[Int](1)(testKit.system.classicSystem)
+      val projection = ProjectionBehaviourTestProjection(Source.fromPublisher(testPublisher), testProbe)
+
+      val projectionRef = testKit.spawn(ProjectionBehavior(projection))
+
+      testProbe.expectMessage(StartObserved)
+
+      testPublisher.sendNext(1)
+      testProbe.expectMessage(Consumed(1, "1"))
+      testPublisher.sendNext(2)
+      testProbe.expectMessage(Consumed(2, "1-2"))
+
+      // stop the projection and wait for the actor to terminate
+      projectionRef ! ProjectionBehavior.Stop
+      testProbe.expectMessage(StopObserved)
+      testProbe.expectTerminated(projectionRef)
+
+      // Respawn the projection behavior
+      testKit.spawn(ProjectionBehavior(projection))
+      testProbe.expectMessage(StartObserved)
+
+      // Assert then projection resumes from where it left
+      testPublisher.sendNext(3)
```

Comment:
It's probably not possible to use `TestPublisher.probe` like this. The second projection will materialize the source again and then I doubt that the original testPublisher will work with that new source.

Reply:
Fair enough. Any suggestions on how to test this aspect at unit level then?
Also, [here](https://github.com/claudio-scandura/akka-projections-playground/blob/master/src/test/scala/com/mylaesoftware/projections/KafkaToSqlProjectionSpec.scala#L152) is another test that does the same thing but with a more realistic projection using Kafka and Postgres as source and sink respectively.

Reply:
I improved the test that verifies that it can resume from stored Kafka offset: https://github.com/akka/akka-projection/pull/442

That PR is also temporarily showing that a ProjectionBehavior can be stopped and started again.

Reply:
That test is using Slick, so it could still be something wrong with Kafka offset storage for JDBC.